README.md

```diff
@@ -212,61 +179,178 @@ animal 2 dalmatian giantpanda cow pig
+$ cat assets/flights_dirty.10.csv
+tuple_id,src,flight,sched_dep_time,act_dep_time,sched_arr_time,act_arr_time
+1,aa,AA-3859-IAH-ORD,7:10 a.m.,7:16 a.m.,9:40 a.m.,9:32 a.m.
+2,aa,AA-1733-ORD-PHX,7:45 p.m.,7:58 p.m.,10:30 p.m.,
+3,aa,AA-1640-MIA-MCO,6:30 p.m.,,7:25 p.m.,
+4,aa,AA-518-MIA-JFK,6:40 a.m.,6:54 a.m.,9:25 a.m.,9:28 a.m.
+5,aa,AA-3756-ORD-SLC,12:15 p.m.,12:41 p.m.,2:45 p.m.,2:50 p.m.
+6,aa,AA-204-LAX-MCO,11:25 p.m.,,12/02/2011 6:55 a.m.,
+7,aa,AA-3468-CVG-MIA,7:00 a.m.,7:25 a.m.,9:55 a.m.,9:45 a.m.
+8,aa,AA-484-DFW-MIA,4:15 p.m.,4:29 p.m.,7:55 p.m.,7:39 p.m.
+9,aa,AA-446-DFW-PHL,11:50 a.m.,12:12 p.m.,3:50 p.m.,4:09 p.m.
 ```
 
-Here, the first entry is the name of the domain, the second entry is a
-unique integer for the cluster index, and the remaining entries are names
-of entities within the domain that belong to this cluster. Recall that the
-schema file `assets/animals.unary.schema` has only one domain, so all the
-lines in the `irm` section start with `animal`.
-
-## Citation
-
-To cite this work, please use the following BibTeX.
-
-```bibtex
-@inproceedings{saad2021hirm,
-title           = {Hierarchical Infinite Relational Model},
-author          = {Saad, Feras A. and Mansinghka, Vikash K.},
-booktitle       = {UAI 2021: Proceedings of the 37th Conference on Uncertainty in Artificial Intelligence},
-fseries         = {Proceedings of Machine Learning Research},
-year            = 2021,
-location        = {Online},
-publisher       = {AUAI Press},
-address         = {Arlington, VA, USA},
-}
+It is okay if the CSV file has columns that don't correspond to queries
+(like `tuple_id` in this example), but currently we don't support schemas
+with queries that don't appear as columns.
+
+Finally, the output of `pclean` is in the same format as `hirm`.  If we look
+at it
+
 ```
+$ head /tmp/flights.out
+3 Time:time src
+4 TrackingWebsite:name
+6 Flight:flight_id
+9 sched_dep_time
+10 act_dep_time flight act_arr_time
+12 sched_arr_time
+
+```
+
+we can see that it clustered the relations into six clusters.  If we look
+at the cluster named `10` in the output file:
+
+```
+irm=10
+TrackingWebsite 1 1 3 6 7 8
+TrackingWebsite 2 0 2 4 5
+Obs 1 0 1 3 4 5 6 7 8
+Obs 2 2
+Flight 0 0 2 3 4 6 7 8
+Flight 1 1 5
+Time 0 0 1 2 3 4 5 6 7 8
+```
+
+we can see that it clustered the underlying entities into three clusters
+(named 0, 1, and 2).  All of the rows of the CSV file (which correspond
+to the `Obs` domain, because of the `from Obs` in the schema) were put
+into cluster 1 except for the line
+
+```
+3,aa,AA-1640-MIA-MCO,6:30 p.m.,,7:25 p.m.,
+```
+
+which was put into cluster 2, probably because it was missing two fields.
+All of the flights were put into cluster 0, except for "AA-1733-ORD-PHX"
+and "AA-204-LAX-MCO", which were put into cluster 1.
+
+Run `./bazel-bin/pclean/pclean --help` for more information about the
+other options that it supports.
 
 ## License
 
-Copyright (c) 2021 MIT Probabilistic Computing Project
+Copyright (c) 2024
```

Comment:
(Not super important): Do you know if there is something special we should do (i.e. is there something we need to adjust because googlers are doing this, or that this was derived from probsys).

Reply:
I'll ask Rif.